### PR TITLE
Cli

### PR DESCRIPTION
**Pull request by chinnaxs**
Cli and HTTP client for beer_go API
./cmd get
./cmd get <beername>
./cmd put <beername> <desc> <price>
./cmd del <beername>

### CODE REVIEW

File: internal/pkg/api/beer.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package api
```
Comment (fehlmach) [inline internal/pkg/api/beer.go]: to keep it consistent. Have your own package beverage.

File: internal/pkg/cmd/put.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+var putCmd = &cobra.Command{
+	Use:   "put [name, description, price]",
+	Short: "Put more beer in the store",
+	Args:  cobra.ExactArgs(3),
+	Run: func(cmd *cobra.Command, args []string) {
+		runPut(cmd, args)
+	},
+}
+
+func runPut(cmd *cobra.Command, args []string) {
```
Comment (fehlmach) [inline internal/pkg/cmd/put.go]: call it addBeer

File: internal/pkg/cmd/put.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	beer := &api.Beer{args[0], args[1], price}
+	err = api.UpdateBeer(c, beer)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	fmt.Printf("succesfully put beer %s\n", args)
+}
+
+func init() {
```
Comment (fehlmach) [inline internal/pkg/cmd/put.go]: put this directly after import

File: internal/pkg/cmd/get.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/chinnaxs/go_beer_client/internal/pkg/api"
+	"github.com/spf13/cobra"
+)
+
+var getCmd = &cobra.Command{
+	Use:   "get [opt:beername",
```
Comment (fehlmach) [inline internal/pkg/cmd/get.go]: missing bracket?

File: internal/pkg/cmd/get.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	Short: "get beer from the store",
+	Long:  `Lists all beers available if no parameter supplied. Gets details of single beer if name is supplied as parameter.`,
+	Args:  cobra.MaximumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		runGet(cmd, args)
+	},
+}
+
+func runGet(cmd *cobra.Command, args []string) {
```
Comment (fehlmach) [inline internal/pkg/cmd/get.go]: name: getBeers

File: internal/pkg/cmd/get.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	var c = &http.Client{}
+	if len(args) > 0 {
+		beerName := args[0]
```
Comment (fehlmach) [inline internal/pkg/cmd/get.go]: put if block in separate method

File: internal/pkg/api/client.go [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+func GetBeer(c *http.Client, name string) (*Beer, error) {
+	url := fmt.Sprintf("%s/%s", baseUrl, name)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := c.Do(req)
+	if err != nil {
+		return nil, err
+	}
```
Comment (fehlmach) [inline internal/pkg/api/client.go]: same as above

File: internal/pkg/api/client.go [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
```
Comment (fehlmach) [inline internal/pkg/api/client.go]: use http.StatusOK

File: internal/pkg/api/client.go [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+	req, err := http.NewRequest("PUT", url, bytes.NewReader(putBody))
+	if err != nil {
+		return err
+	}
+	resp, err := c.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
```
Comment (fehlmach) [inline internal/pkg/api/client.go]: response can be 201 Created or 204 no content

File: internal/pkg/api/client.go [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+	req, err := http.NewRequest("DELETE", url, nil)
+	if err != nil {
+		return err
+	}
+	resp, err := c.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
```
Comment (fehlmach) [inline internal/pkg/api/client.go]: response is 204 -> http.StatusNoContent

File: internal/pkg/cmd/put.go (hunk 2 above)
Comment (chinnaxs) [reply internal/pkg/cmd/put.go]: called it addOrUpdateBeer

File: internal/pkg/cmd/put.go (hunk 3 above)
Comment (chinnaxs) [reply internal/pkg/cmd/put.go]: doone

File: internal/pkg/cmd/get.go (hunk 5 above)
Comment (chinnaxs) [reply internal/pkg/cmd/get.go]: called it listBeer

File: internal/pkg/cmd/del.go [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+var delCmd = &cobra.Command{
+	Use:   "del [name]",
+	Short: "delete beer from the store",
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		runDel(cmd, args)
+	},
+}
+
+func runDel(cmd *cobra.Command, args []string) {
```
[the remark this answers is not in the document]
Comment (chinnaxs) [reply internal/pkg/cmd/del.go]: called it removeBeer

**Pull request merged**